### PR TITLE
feat(sync): Send migrate messages

### PR DESCRIPTION
**Pull request by erezrokah**
First part of the sync, sending migrate table messages to the client

### CODE REVIEW

File: lib/src/main/java/io/cloudquery/internal/servers/plugin/v3/PluginServer.java [hunk 1]
```diff
@@ -64,18 +58,7 @@ public void getTables(
               request.getSkipDependentTables());
       List<ByteString> byteStrings = new ArrayList<>();
       for (Table table : tables) {
-        try (BufferAllocator bufferAllocator = new RootAllocator()) {
```
Comment (erezrokah) [inline lib/src/main/java/io/cloudquery/internal/servers/plugin/v3/PluginServer.java:67]: Moved the the `Table` class under the `encode` method

File: lib/src/main/java/io/cloudquery/plugin/BackendOptions.java [hunk 2]
```diff
@@ -0,0 +1,11 @@
+package io.cloudquery.plugin;
+
+import lombok.AllArgsConstructor;
+import lombok.Getter;
+
+@AllArgsConstructor
+@Getter
+public class BackendOptions {
```
Comment (erezrokah) [inline lib/src/main/java/io/cloudquery/plugin/BackendOptions.java:8]: Wrapper class so consumers don't need to mess with the gRPC wrapper

File: lib/src/main/java/io/cloudquery/server/ServeCommand.java [hunk 3]
```diff
@@ -88,6 +88,7 @@ private LoggerContext initLogger() {
     context.start(configuration);
 
     logger = context.getLogger(ServeCommand.class.getName());
+    this.plugin.setLogger(logger);
```
Comment (erezrokah) [inline lib/src/main/java/io/cloudquery/server/ServeCommand.java:91]: Setting the logger on the plugin was missing

File: lib/src/main/java/io/cloudquery/schema/Table.java [hunk 4]
```diff
@@ -228,4 +236,19 @@ public Schema toArrowSchema() {
     Schema schema = new Schema(asList(fields), metadata);
     return schema;
   }
+
+  public ByteString encode() throws IOException {
+    try (BufferAllocator bufferAllocator = new RootAllocator()) {
```
Comment (erezrokah) [inline lib/src/main/java/io/cloudquery/schema/Table.java:241]: Moved from `PluginServer.java`

**Review by mnorbury — APPROVED** [review]

**Pull request merged**